### PR TITLE
Cleanup of SMT2 solver test execution and CMake support [blocks: #4063]

### PR DESCRIPTION
**Pull request by tautschnig**
I'd suggest reviewing commit-by-commit.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: b68b68bc081296e2cf83d4c7981266e1aed80316).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100124409
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 9632c6b09e63a210adcd34ff5e43b924f7cdbd45).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100143902

File: regression/cbmc/Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ test:
 	@../test.pl -p -c "../../../src/cbmc/cbmc --validate-goto-model" -X smt-backend
 
 test-cprover-smt2:
-	@../test.pl -p -c "../../../src/cbmc/cbmc --cprover-smt2"
+	@../test.pl -p -c "../../../src/cbmc/cbmc --cprover-smt2" -X broken-smt-backend
```
Comment (romainbrenguier) [inline regression/cbmc/Makefile:7]: The commits are in the wrong order, this should go after the broken-smt-backend tag is introduced

File: regression/cbmc/Anonymous_Struct3/test.desc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CORE
+CORE broken-smt-backend
```
Comment (romainbrenguier) [inline regression/cbmc/Anonymous_Struct3/test.desc:1]: typo in commit message? "easier to repeated build" -> "easier to repeatedly build"? 
"regression test" -> "regression tests"?

File: regression/cbmc/CMakeLists.txt [hunk 3]
```diff
@@ -8,3 +8,15 @@ add_test_pl_profile(
     "-C;-X;thorough-paths;-X;smt-backend;-s;paths-lifo"
     "CORE"
 )
+
```
Comment (romainbrenguier) [inline regression/cbmc/CMakeLists.txt:11]: typo in commit message: "partity"

File: regression/cbmc/Makefile (hunk 1 above)
Comment (tautschnig) [reply regression/cbmc/Makefile:7]: I think that's just GitHub displaying them in the wrong order, but let me make this unambiguous by moving this particular change into the same commit that introduces the tag.

File: regression/cbmc/Anonymous_Struct3/test.desc (hunk 2 above)
Comment (tautschnig) [reply regression/cbmc/Anonymous_Struct3/test.desc:1]: Thanks a lot!

File: regression/cbmc/CMakeLists.txt (hunk 3 above)
Comment (tautschnig) [reply regression/cbmc/CMakeLists.txt:11]: Thanks, fixed!

File: regression/cbmc/Makefile (hunk 1 above)
Comment (romainbrenguier) [reply regression/cbmc/Makefile:7]: You can update the commit date to make github display them in the right order `git rebase "$(git merge-base HEAD origin/develop)" --ignore-date -x 'git commit --amend -C HEAD --date="$(date -R)" && sleep 1.05' -i`

**Review by romainbrenguier — APPROVED** [review]
Should wait for CI but it's a good thing to do

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 3fc215c88e95d7ae9683f4934075434f3eddcda3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100235492

**Review by smowton — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**